File: java/prob1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> myMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            myMap.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (myMap.containsKey(diff) && myMap.get(diff) != i) 
                return new int[] { i, myMap.get(diff)};
        }
        return null;
    }
}

File: python3/prob1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        d = dict((nums[i], i) for i in range (len(nums)))
        for i in range (len(nums)):
            diff = target - nums[i]
            if diff in d.keys() and d.get(diff) != i:
                 return [i, d.get(diff)]
        
